server/sockets.js

```diff
@@ -404,7 +405,7 @@ if (cluster.isMaster) {
 			// @ts-ignore
 			options.faye_server_options = {extensions: [deflate]};
 		} catch (e) {
-			require('../lib/crashlogger')(new Error("Dependency permessage-deflate is not installed or is otherwise unaccessable. No message compression will take place until server restart."), "Sockets");
+			Monitor.crashlog(new Error("Dependency permessage-deflate is not installed or is otherwise unaccessable. No message compression will take place until server restart."), "Sockets");
```

Comment:
Is `Monitor.crashlog` actually defined in this process? You might want to check that before replacing the rest of these. Most of the ones that aren't already `Monitor.crashlog`, haven't been replaced because I haven't gotten around to doing that yet.

Reply:
Apparently not: `./pokemon-showdown start` failed. I've fixed the locations such that it now starts without errors again - maybe there should be an integration test with runs `pokemon-showdown start` and confirms it runs without errors?

Reply:
Integration tests are for losers!

Also we intentionally avoid doing that because it could conflict with existing ports being listened to, and also write things to logs we don't want written.

Reply:
Honestly, we _should_ have an integration test for `./pokemon-showdown simulate-battle`, because that one's actually self-contained.

Reply:
I mean, I'm feeling embarrassed I [crashed the server](https://github.com/Zarel/Pokemon-Showdown/compare/5de019ac3d28...13accb33eb3d) recently and am wondering how we can improve test coverage/canarying etc. I mean, in the end it wasn't an issue because you patched quickly, and we're "Beta" and what not, but I still want to know if we can do better. I also wonder if we might be able to do something like run the OMs/mods in their own process so [they can't mess up the core experience](https://www.smogon.com/forums/threads/ps-serious-bug-reports.3521891/page-14#post-7955555).

This seems out of context for this bug. I will follow a followup bug for discussion, and at the very least we can have a `./pokemon-showdown simulate-battle` test.

Reply:
I'd definitely support the idea of running OMs in their own process.

Reply:
Great.